Add Exposure Alert UI

johnsonh opened the pull request:
- AlertCell, which creates `AlertView` from nib using a generic extension https://stackoverflow.com/a/36388769
- Set table view size to 360 x 700, table view rows and `AlertView` to 360 x 120
- Hardcoded model, hook up model and view with Rx

![Simulator Screen Shot - iPhone 11 Pro - 2020-03-29 at 20 51 17](https://user-images.githubusercontent.com/5308839/77849585-6b5bd100-71ff-11ea-8203-d0c5e56774e5.png)

TODO:
- Acknowledge button behavior - blocked on UX discussion (what should happen when acknowledge is pressed?)
- Fonts are wrong - blocked on https://github.com/Co-Epi/app-ios/pull/23
- Icon is wrong - blocked on UX discussion (The icon is a radar? Is it static? Does it correspond to the exposure?)
- Label text is hardcoded - blocked on UX discussion (What are we displaying? Symptoms? "Severity of exposure?" Like how close you came to them?)

i-schuetz commented on CoEpi/ui/alerts/AlertCell.swift:
```diff
@@ -1,3 +1,40 @@
+    private func setupUI(alert: Alert) {
+        let view: AlertView = AlertView.fromNib()
+        view.onAcknowledged = { [weak self] in
+            guard let this = self, let alert = this.alert else { return }
```
You don't need these, just
```swift
view.onAcknowledged = { [weak self] in
    self?.onAcknowledged?(alert)
}
```
Also, let's use `let self = self`, for consistency

i-schuetz commented on CoEpi/ui/alerts/AlertCell.swift:
```diff
@@ -1,3 +1,40 @@
+import UIKit
 
+class AlertCell: UITableViewCell {
+    private static let nibName = "AlertView"
```
Not used

i-schuetz commented on CoEpi/ui/alerts/AlertCell.swift:
```diff
@@ -1,3 +1,40 @@
+    }
+}
+
+extension UIView {
```
Put this in the ui extensions folder.

i-schuetz commented on CoEpi/ui/alerts/AlertsViewController.swift:
```diff
@@ -7,26 +7,70 @@
+import RxCocoa
 
 class AlertsViewController: UIViewController {
+    private let viewModel: AlertsViewModel = .init()
```
The view models should still be injected, since they can have more dependencies.

i-schuetz commented on CoEpi/ui/alerts/AlertsViewController.swift:
```diff
@@ -7,26 +7,70 @@
+        contactAlerts.register(cellClass: AlertCell.self)
+
+        viewModel.alerts.asObservable()
+            .subscribe(onNext: { [unowned self] alerts in
```
Let's avoid `unowned`. It doesn't really bring a benefit and makes the code more error prone (e.g. in refactorings).

i-schuetz commented on CoEpi/ui/alerts/AlertsViewController.swift:
```diff
@@ -7,26 +7,70 @@
+        if count < 2 {
+            return title
+        }
+        return title + "s"
```
Move this to the view model. The view controller should observe only a `Driver` with the text.

i-schuetz commented on CoEpi/ui/alerts/AlertsViewModel.swift:
```diff
@@ -1,3 +1,21 @@
+            Alert(id: "b", exposure: "testb"),
+            Alert(id: "c", exposure: "testc"),
+            Alert(id: "d", exposure: "testd"),
+        ]
```
This should be in a repo

i-schuetz commented on CoEpi/ui/alerts/AlertsViewModel.swift:
```diff
@@ -1,3 +1,21 @@
+            Alert(id: "c", exposure: "testc"),
+            Alert(id: "d", exposure: "testd"),
+        ]
+        alerts = BehaviorRelay(value: tempAlerts)
```
There's no reason to use `BehaviorRelay` here. To return a fixed value, use `Observable.just(value)`

i-schuetz commented:
Also, see with dark mode:
![IMG_0562](https://user-images.githubusercontent.com/1381744/77852392-b1546d00-71de-11ea-80ac-2959e6f1a97e.PNG)

Are you using constraints in the xib to pin the edges of the table view to the parent? I remember the symptoms also looked like this.

johnsonh commented:
Fixed in dark mode and addressed comments
![Screen Shot 2020-03-30 at 12 19 32 AM](https://user-images.githubusercontent.com/5308839/77854321-314cf800-721c-11ea-8885-6e7664c34694.png)

duskoo commented on CoEpi/ui/alerts/AlertsViewController.swift:
```diff
@@ -7,26 +7,73 @@
+
+        alertCell.onAcknowledged = { [weak self] (alert) in
+            // TODO: what happens in the UI when an alert is acknowledged?
+            // TODO: pipe to VM
```
We should at least log something here.

i-schuetz commented on CoEpi/ui/alerts/AlertsViewController.swift:
```diff
@@ -7,26 +7,73 @@
+        viewModel.alerts.asObservable()
+            .subscribe(onNext: { [weak self] alerts in
+                guard let self = self else { return }
+                self.titleLabel.text = self.viewModel.formatTitleLabel(count: alerts.count)
```
```swift
viewModel.title
    .drive(titleLabel.rx.text)
    .disposed(by: disposeBag)
```

johnsonh replied:
Ah much cleaner 👍
